### PR TITLE
feat: added codecov, golangci-lint, security scan

### PR DESCRIPTION
**Pull request by psturc**
https://issues.redhat.com/browse/HACBS-1055

### CODE REVIEW

File: .github/workflows/codecov-main.yaml
Comment (psturc) [reply .github/workflows/codecov-main.yaml:1]: codecov is used in many [openshift org's repositories](https://app.codecov.io/gh/openshift) and also in [redhat-appstudio's](https://app.codecov.io/gh/redhat-appstudio).

I do not know whether there's a strict requirement to use only this tool.
However the codecov suggestions/warning do not force developers to address all of them, it's up to an individual person to decide whether it makes sense to take some actions on them.

The main purpose of using this tool is to provide an information about a test coverage and suggest how it can be improved. Also there are ways to configure this tool, i.e. possibly we might be able to remove the codecov code suggestions in case they prove to be too much annoying :)

Comment (psturc) [reply .github/workflows/codecov-main.yaml:1]: golanci-lint - I will try to address it in this PR

**Comment by gabemontero** [comment]
the gosec complaints are "won't fix"as I see it, as what @stuartwdouglas is doing is not encrypting data, but rather just encoding 

the test result page seemed to provide a way of indicating that, though I'm curious of that selection persists across PRs @psturc

**Comment by psturc** [comment]
> the gosec complaints are "won't fix"as I see it, as what @stuartwdouglas is doing is not encrypting data, but rather just encoding
> 
> the test result page seemed to provide a way of indicating that, though I'm curious of that selection persists across PRs @psturc

I think we can remove it by [annotating those pieces of code](https://github.com/securego/gosec#annotating-code) that we know that aren't a security issue

**Comment by gabemontero** [comment]
> > the gosec complaints are "won't fix"as I see it, as what @stuartwdouglas is doing is not encrypting data, but rather just encoding
> > the test result page seemed to provide a way of indicating that, though I'm curious of that selection persists across PRs @psturc
> 
> I think we can remove it by [annotating those pieces of code](https://github.com/securego/gosec#annotating-code) that we know that aren't a security issue

cool ... so as I read https://github.com/redhat-appstudio/jvm-build-service/pull/200/checks?check_run_id=8093189957:
- for dependencybuild.go, that is currently line 5, 155, 593
- for artifacdtbuild.go, that is currently line 5, 6, 376, 499
can you handle those as well with this PR @psturc ?  or would you rather defer to @stuartwdouglas or I for that ?

**Comment by psturc** [comment]
> > > the gosec complaints are "won't fix"as I see it, as what @stuartwdouglas is doing is not encrypting data, but rather just encoding
> > > the test result page seemed to provide a way of indicating that, though I'm curious of that selection persists across PRs @psturc
> > 
> > 
> > I think we can remove it by [annotating those pieces of code](https://github.com/securego/gosec#annotating-code) that we know that aren't a security issue
> 
> cool ... so as I read https://github.com/redhat-appstudio/jvm-build-service/pull/200/checks?check_run_id=8093189957:
> 
> * for dependencybuild.go, that is currently line 5, 155, 593
> * for artifacdtbuild.go, that is currently line 5, 6, 376, 499
>   can you handle those as well with this PR @psturc ?  or would you rather defer to @stuartwdouglas or I for that ?

I'm on it - will add a comment here once it's ready for a review

**Comment by psturc** [comment]
issues with CI
```
Connecting to github.com (github.com)|140.82.112.3|:443... connected.
HTTP request sent, awaiting response... 500 Internal Server Error
2022-08-30 15:18:16 ERROR 500: Internal Server Error.
```

/test jvm-build-service-e2e
/test jvm-build-service-in-repo-with-pruning-e2e
/test jvm-build-service-in-repo-e2e

**Comment by psturc** [comment]
all tests passed 😮 

@stuartwdouglas @gabemontero  ready for another round of reviews

File: openshift-with-appstudio-test/e2e/types.go
Comment (gabemontero) [inline openshift-with-appstudio-test/e2e/types.go]: interesting; I didn't need this to build

any idea why you needed it @psturc ?

File: openshift-with-appstudio-test/e2e/util.go
Comment (gabemontero) [inline openshift-with-appstudio-test/e2e/util.go:55]: ioutil was flagged for some reason then?

File: pkg/reconciler/artifactbuild/artifactbuild.go
Comment (gabemontero) [inline pkg/reconciler/artifactbuild/artifactbuild.go:507]: interesting ... this give you any pause @stuartwdouglas ?  I think it is OK but this name generation is yours

File: pkg/reconciler/dependencybuild/dependencybuild.go
Comment (gabemontero) [reply pkg/reconciler/dependencybuild/dependencybuild.go:594]: yeah that is "slice 101" ... good catch on gosec's part

**Review by gabemontero — COMMENTED** [review]
I couple of minor questions @psturc but seems good to me

Aside from answering those, let's get a yay/nay from @stuartwdouglas before merging

thanks!!

File: openshift-with-appstudio-test/e2e/util.go
Comment (psturc) [reply openshift-with-appstudio-test/e2e/util.go:55]: yes, [deprecated from v1.16](https://pkg.go.dev/io/ioutil)

**Review by stuartwdouglas — APPROVED** [review]

File: openshift-with-appstudio-test/e2e/types.go
Comment (psturc) [reply openshift-with-appstudio-test/e2e/types.go]: apologies, I should have mentioned that golangci-lint wasn't originally including [jvm_build_service_test.go](https://github.com/redhat-appstudio/jvm-build-service/blob/main/openshift-with-appstudio-test/e2e/jvm_build_service_test.go) and [periodic_test.go](https://github.com/redhat-appstudio/jvm-build-service/blob/main/openshift-with-appstudio-test/e2e/periodic_test.go) (because of the build tags), hence it evaluated types and functions in go files in this folder as unused. So I updated these files with build tags and they mean that they're going to be included into go build/run only if we specify `normal` or `periodic` tag (which is already set in [Makefile](https://github.com/redhat-appstudio/jvm-build-service/blob/c5469a6c6037922b9a2dfc28b69b856ebd0c46f6/Makefile#L33-L37))

Comment (psturc) [reply openshift-with-appstudio-test/e2e/types.go]: in fact the way I configured golangci-lint and gosec jobs is wrong. I created a follow-up PR with fixes and small improvements: https://github.com/redhat-appstudio/jvm-build-service/pull/207

**Pull request merged**
